src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalClusterExtension.java

```diff
@@ -108,13 +107,12 @@ public void beforeAll(ExtensionContext context) {
 	private UnaryOperator<Neo4jContainer<?>> findContainerModifier(Class<?> requiredTestClass, Object testInstance,
 		Class<? extends Annotation> annotationType) {
 
-		final Predicate<Method> methodFilter = testInstance == null ?
-			method -> Modifier.isStatic(method.getModifiers())
-			: method -> true;
-
 		return AnnotationSupport.findAnnotatedMethods(requiredTestClass, annotationType, TOP_DOWN).stream()
-			.filter(methodFilter)
 			.map(method -> {
+				if (testInstance == null && !Modifier.isStatic(method.getModifiers())) {
+					throw new IllegalArgumentException(
+						"Instance methods are not supported for @" + annotationType.getSimpleName());
+				}
```

Comment:
before if you annotated a non-static method on a static test class you would think that your config was being changed but the method was just silently ignored.
Now it throws an error, to avoid the ambiguity & confusion.